### PR TITLE
update dependency struct to include optional total_count

### PR DESCRIPTION
**Pull request by energyvault462**

### CODE REVIEW

File: scans/results.go [hunk 1]
```diff
@@ -296,6 +296,7 @@ type Dependency struct {
 	Requirement   string       `json:"requirement" xml:"requirement"`
 	File          string       `json:"file" xml:"file"`
 	Dependencies  []Dependency `json:"dependencies" xml:"dependencies"`
+	TotalCount    int          `json:"total_count,omitempty"`
```
Comment (kellyp) [inline scans/results.go:299]: How is this different that the length of the dependencies?

Comment (energyvault462) [reply scans/results.go:299]: it's the same dependency in different projects. Would renaming it make more sense? ProjectCount perhaps?

Comment (kellyp) [reply scans/results.go:299]: You probably want a new package/struct for what you are doing.  The data here is for the results of a scan.

Comment (energyvault462) [reply scans/results.go]: removed the reports edit and created it in portfolio package where it is used.

**Pull request merged**